backend/src/main/java/ru/hh/superscoring/service/TestPassService.java

```diff
@@ -0,0 +1,35 @@
+package ru.hh.superscoring.service;
+
+import java.util.List;
+import ru.hh.superscoring.dao.TestPassRecordDao;
+import ru.hh.superscoring.entity.TestPassRecord;
+
+public class TestPassService {
+  private final TestPassRecordDao testPassRecordDao;
+  private final QuestionService questionService;
+
+  public TestPassService(TestPassRecordDao testPassRecordDao, QuestionService questionService) {
+    this.testPassRecordDao = testPassRecordDao;
+    this.questionService = questionService;
+  }
+
+  public void saveRecord(Integer testID, Integer userId) {
+    if (!testPassRecordDao.isExistUnfinishedRecord(userId)) {
+      TestPassRecord rec = new TestPassRecord();
+      rec.setTestId(testID);
+      rec.setUserId(userId);
+      rec.setQuestionIds(questionService.getQuestionsForStart(testID, 10));
+      System.out.println(rec.getQuestionIds());
+      testPassRecordDao.save(rec);
+    } else {
+      System.out.println("Для этого пользователя уже существует запись о незавершенном тестированиии.");
+    }
+  }
+
+  public List<Integer> getUnfinishedPassQuestions(Integer userId) {
+    if (testPassRecordDao.isExistUnfinishedRecord(userId)) {
+      return testPassRecordDao.getTestPassQuestionsByUser(userId);
+    }
+    return null;
```

Comment:
```suggestion
    return List.of();
```
Лучше возвращать в заявленном в методе виде

Reply:
ок